src/App.tsx

```diff
@@ -15,33 +15,140 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App: React.FC = () => {
+  const goods = [...goodsFromServer];
+
+  const [isVisible, setVisio] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [isSortByAlphabet, sortAlphabetically] = useState(false);
+  const [isSortByLength, sortByLength] = useState(false);
+  const [isReset, setResetedGoods] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goods.filter(good => good.length >= Number(goodsLength));
+
+  switch (true) {
+    case isSortByLength:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case isSortByAlphabet:
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  if (isReset) {
+    setReversed(false);
+    sortAlphabetically(false);
+    sortByLength(false);
+    setResetedGoods(false);
+    setGoodsLength('1');
+  }
+
+  return (
+    <div className="App message">
+      {!isVisible && (
+        <div>
+          <p
+            className="title message-header"
+          >
+            Control of goods starts from pushing this
+          </p>
+          <button
+            type="button"
+            className="button is-large is-fullwidth"
+            onClick={() => setVisio(true)}
+          >
+            Start
+          </button>
+        </div>
+      )}
+
+      {isVisible && (
+        <div>
+          <button
+            type="button"
+            className="button is-dark"
+            onClick={() => {
+              sortAlphabetically(true);
+              sortByLength(false);
+              setReversed(false);
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className="button is-dark"
+            onClick={() => {
+              sortByLength(true);
+              sortAlphabetically(false);
+              setReversed(false);
+            }}
```

Comment:
this should not be written in place, you'd rather create a separate function-handler, and then just pass it right here